### PR TITLE
fix(audit): provide correct leveled_logger for retryablehttp client

### LINKED ISSUE #3283: [Bug] panic when using audit webhook
### Bug Description

Got panic when using flipt with audit webhook is configured. This cause server unavailable and audit function does not work properly

### Version Info

v1.46.0

### Search

- [X] I searched for other open and closed issues before opening this

### Steps to Reproduce

Config follow https://github.com/flipt-io/flipt/blob/v1.46.0/examples/audit/webhook/docker-compose.yml
When create flag via UI, got below panic and process is not reachable anymore

```
webhook-flipt-1    | panic: invalid logger type passed, must be Logger or LeveledLogger, was *zap.Logger
webhook-flipt-1    | 
webhook-flipt-1    | goroutine 135 [running]:
webhook-flipt-1    | github.com/hashicorp/go-retryablehttp.(*Client).logger.func1()
webhook-flipt-1    | 	github.com/hashicorp/go-retryablehttp@v0.7.7/client.go:463 +0xcd
webhook-flipt-1    | sync.(*Once).doSlow(0x487c40?, 0xc0000f33f0?)
webhook-flipt-1    | 	sync/once.go:74 +0xc2
webhook-flipt-1    | sync.(*Once).Do(...)
webhook-flipt-1    | 	sync/once.go:65
webhook-flipt-1    | github.com/hashicorp/go-retryablehttp.(*Client).logger(0xc0000f3380)
webhook-flipt-1    | 	github.com/hashicorp/go-retryablehttp@v0.7.7/client.go:453 +0x45
webhook-flipt-1    | github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc0000f3380, 0xc000acea08)
webhook-flipt-1    | 	github.com/hashicorp/go-retryablehttp@v0.7.7/client.go:656 +0x9b
webhook-flipt-1    | go.flipt.io/flipt/internal/server/audit/webhook.(*webhookClient).SendAudit(0xc0008c4ff0, {0x3b16400, 0xc000372e70}, {{0x304c485, 0x3}, {0x304cd89, 0x4}, {0x305204e, 0x7}, {0xc0008b6730}, ...})
webhook-flipt-1    | 	go.flipt.io/flipt/internal/server/audit/webhook/client.go:72 +0x32f
webhook-flipt-1    | go.flipt.io/flipt/internal/server/audit/webhook.(*Sink).SendAudits(0xc000b32fd8, {0x3b16400, 0xc000372e70}, {0xc000372f50?, 0x2?, 0x2?})
webhook-flipt-1    | 	go.flipt.io/flipt/internal/server/audit/webhook/webhook.go:31 +0x119
webhook-flipt-1    | go.flipt.io/flipt/internal/server/audit.(*SinkSpanExporter).SendAudits(0xc0008d4d40, {0x3b16400, 0xc000372e70}, {0xc000372f50, 0x1, 0x1})
webhook-flipt-1    | 	go.flipt.io/flipt/internal/server/audit/audit.go:85 +0x2e9
webhook-flipt-1    | go.flipt.io/flipt/internal/server/audit.(*SinkSpanExporter).ExportSpans(0xc0008d4d40, {0x3b16400, 0xc000372e70}, {0xc0008d4d60?, 0xc000b2bef2?, 0xc0009c2540?})
webhook-flipt-1    | 	go.flipt.io/flipt/internal/server/audit/audit.go:60 +0xd9
webhook-flipt-1    | go.opentelemetry.io/otel/sdk/trace.(*batchSpanProcessor).exportSpans(0xc0006f61e0, {0x3b16390, 0xc000858140})
webhook-flipt-1    | 	go.opentelemetry.io/otel/sdk@v1.27.0/trace/batch_span_processor.go:277 +0x238
webhook-flipt-1    | go.opentelemetry.io/otel/sdk/trace.(*batchSpanProcessor).processQueue(0xc0006f61e0)
webhook-flipt-1    | 	go.opentelemetry.io/otel/sdk@v1.27.0/trace/batch_span_processor.go:321 +0x308
webhook-flipt-1    | go.opentelemetry.io/otel/sdk/trace.NewBatchSpanProcessor.func1()
webhook-flipt-1    | 	go.opentelemetry.io/otel/sdk@v1.27.0/trace/batch_span_processor.go:117 +0x54
webhook-flipt-1    | created by go.opentelemetry.io/otel/sdk/trace.NewBatchSpanProcessor in goroutine 1
webhook-flipt-1    | 	go.opentelemetry.io/otel/sdk@v1.27.0/trace/batch_span_processor.go:115 +0x2e5
```

### Expected Behavior

Webhook should be fired and server still be healthy

### Additional Context

Docker

### PR DESCRIPTION
**Pull request by erka**
resolves #3283

### CODE REVIEW

File: internal/server/audit/webhook/client.go [hunk 1]
```diff
@@ -40,7 +42,9 @@ func (w *webhookClient) signPayload(payload []byte) []byte {
 }
 
 // NewHTTPClient is the constructor for a HTTPClient.
-func NewWebhookClient(logger *zap.Logger, url, signingSecret string, httpClient *retryablehttp.Client) Client {
+func NewWebhookClient(logger *zap.Logger, url, signingSecret string, maxBackoffDuration time.Duration) Client {
+	httpClient := retryablehttp.NewClient()
+	httpClient.Logger = template.NewLeveledLogger(logger)
```
Comment (GeorgeMac) [inline internal/server/audit/webhook/client.go:47]: Do we need to be setting the RetryWay to maxBackoffDuration here?
```suggestion
	httpClient.Logger = template.NewLeveledLogger(logger)
	httpClient.RetryWait = maxBackoffDuration
```

**Review by GeorgeMac — CHANGES_REQUESTED** [review]
Thanks for fixing this ! Once we get it in, I can cut a patch release.
One outstanding thing I think regarding setting the retry wait ?

File: internal/server/audit/webhook/client.go (hunk 1 above)
Comment (erka) [reply internal/server/audit/webhook/client.go:47]: @GeorgeMac great catch!

**Review by GeorgeMac — APPROVED** [review]
Awesome, thanks again @erka 🙏

**Pull request merged**